api/apiclient.go

```diff
@@ -418,25 +421,27 @@ func (st *state) apiEndpoint(path, query string) (*url.URL, error) {
 	if err != nil {
 		return nil, errors.Trace(err)
 	}
-	return &url.URL{
```

Comment:
This `apiEndpoint()` method is confusing. "api" implies the scheme should be the api scheme "wss". But it's only called from one place `func (s *state) HTTPClient() (*httprequest.Client, error)`
Let's just remove this method and inline the code in that single caller location. It simplifies a lot the cognitive load in understanding it.

Reply:
Inlined and simplified.